src/pages/index.tsx

```diff
@@ -69,8 +72,33 @@ function IndexPage() {
     },
     validate,
     onSubmit: (values) => {
-      console.log(values);
-      setFormSubmitted(true);
+      fetch("http://139.59.154.199:49160/api/v1/leads", {
+        method: "POST",
+        headers: {
+          Authorization: "secret_token",
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          name: values.username,
+          email: values.email,
+          consentsAccepted: values.acceptance,
+        }),
+      })
+        .then((response) => {
+          if (response.ok) {
+            setFormSubmitted(true);
+          } else if (response.status >= 400 && response.status < 500) {
+            setError4xx(true);
```

Comment:
**issue:** Now you hardcoded this error message. For status **500** it makes sense, because it's unhandled server error. For 4xx it usually means that error message is thrown from server with some user friendly message. We should get this error message from server response and show it there, eg:

```
.catch((error) => {
          console.error("Error:", error);
          setErrorMessage(error.message);
        });
```